### PR TITLE
contrib/gonic: new package (0.16.4)

### PR DESCRIPTION
**Pull request by saagraa**

### CODE REVIEW

**Comment by q66** [comment]
please format your template with black or ruff

File: contrib/gonic/template.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+pkgname = "gonic"
+pkgver = "0.16.4"
+pkgrel = 0
+build_style = "go"
+make_build_args = [ "cmd/gonic/gonic.go" ]
+hostmakedepends = [ "go" ]
+makedepends = [ "sqlite-devel", "taglib-devel", "pkgconf" ]
```
Comment (nekopsykose) [inline contrib/gonic/template.py:7]: pkgconf is hostmakedepends

**Comment by nekopsykose** [comment]
this needs a dinit service for it too pretty sure

File: contrib/gonic/template.py (hunk 1 above)
Comment (nekopsykose) [reply contrib/gonic/template.py:7]: you also need go_build_tags = ["libsqlite3"] for sqlite-devel to be used

File: contrib/gonic/template.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+pkgdesc = "Static site generator"
```
Comment (nekopsykose) [inline contrib/gonic/template.py:8]:
```suggestion
pkgdesc = "Subsonic-compatible music streaming server"
```

**Comment by saagraa** [comment]
> this needs a dinit service for it too pretty sure

Should I match the path upstream uses for config (/etc/gonic/config) and leave that file/path nonexistent? A default config doesn't make too much sense for this I think.

**Comment by nekopsykose** [comment]
that sounds like it would make sense, sure

**Comment by nekopsykose** [comment]
> Edit: upstream has a default config, I'll install that

it does, and you could make the `command =` pass -config-path to it, but that default config doesn't work ootb so idk if that makes sense to just put in /etc/ or not

**Comment by nekopsykose** [comment]
for tmpfiles/sysusers, you can also use the upstream contrib one as reference, just name the user `_gonic` instead

File: contrib/gonic/files/gonic [hunk 3]
```diff
@@ -0,0 +1,5 @@
+type = process
+command = /usr/bin/gonic -config-path /etc/gonic/config
+depends-on = network.target
```
Comment (nekopsykose) [inline contrib/gonic/files/gonic:3]: network.target is only supplementary; you want to also add depends-on = local.target

File: contrib/gonic/template.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+pkgname = "gonic"
+pkgver = "0.16.4"
+pkgrel = 0
+build_style = "go"
+make_build_args = ["cmd/gonic/gonic.go"]
```
Comment (nekopsykose) [inline contrib/gonic/template.py:5]:
```suggestion
make_build_args = ["./cmd/gonic"]
```

File: contrib/gonic/template.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+makedepends = ["sqlite-devel", "taglib-devel"]
+go_build_tags = ["libsqlite3"]
+pkgdesc = "Music streaming server / subsonic server API implementation"
+maintainer = "samg <samg@fastmail.com>"
+license = "GPL-3.0-only"
+url = "https://github.com/sentriz/gonic"
+source = f"https://github.com/sentriz/gonic/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "ba690a861a075bdf3f1f7e8131e45a5ca430ec90902a97faf955ec9f36799461"
+# tests require network access, transcoding test requires ffmpeg
+options = ["!check", "!cross"]
```
Comment (nekopsykose) [inline contrib/gonic/template.py:16]:
```suggestion
# taglib gomod can't cross
options = ["!cross"]
```

Comment (nekopsykose) [reply contrib/gonic/template.py:16]: (the default go test ./... seems to work fine)

File: contrib/gonic/template.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+build_style = "go"
+make_build_args = ["cmd/gonic/gonic.go"]
+hostmakedepends = ["go", "pkgconf"]
+makedepends = ["sqlite-devel", "taglib-devel"]
+go_build_tags = ["libsqlite3"]
+pkgdesc = "Music streaming server / subsonic server API implementation"
+maintainer = "samg <samg@fastmail.com>"
+license = "GPL-3.0-only"
+url = "https://github.com/sentriz/gonic"
+source = f"https://github.com/sentriz/gonic/archive/refs/tags/v{pkgver}.tar.gz"
```
Comment (nekopsykose) [inline contrib/gonic/template.py:13]:
```suggestion
source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
```

File: contrib/gonic/template.py (hunk 5 above)
Comment (saagraa) [reply contrib/gonic/template.py:16]: I'm getting a test failure in when building in the sandbox but not outside of it; still trying to sort it out

Comment (nekopsykose) [reply contrib/gonic/template.py:16]: i'm not sure how you're testing that but just running a regular `pkg` with this works fine for me, just push it and see what CI says

Comment (nekopsykose) [reply contrib/gonic/template.py]: hm ok seems to fail, why did it work for me...

Comment (nekopsykose) [reply contrib/gonic/template.py]: and it definitely runs the same failing test for me locally

```
ok  	go.senan.xyz/gonic/server/ctrlsubsonic	0.101s
```

nothing in the files seem to e.g. do something different based on CI in env or whatnot so i'm very confused..

Comment (nekopsykose) [reply contrib/gonic/template.py]: aand now it works

so i guess that's actually a flake that only sometimes happens, oh well

**Pull request merged**